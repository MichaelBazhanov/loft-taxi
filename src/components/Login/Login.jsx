import React from "react";
import LogoTaxiVertical from '../../images/logo-taxi-vertical.svg';

class Login extends React.Component {
	state = { email: '', password: '', }

	handleSubmit = (event) => {
		event.preventDefault()

		// делаем что то с данными
		// console.log(this.state.email, this.state.password)
		alert('Форма отправлена!')
	}

	handleChange = (event) => {
		this.setState({ [event.target.name]: event.target.value })
	}

	render() {
		const { email, password } = this.state

		return (
			<section
				className="container mx-auto flex h-screen">
				<div
					className="w-1/3 bg-black-me flex justify-center items-center">
					<img
						src={LogoTaxiVertical}
						className="logo-taxi-vertical"
						alt="logo" />
				</div>
				<div
					className="w-2/3 flex justify-center items-center">
					<form
						onSubmit={this.handleSubmit} className="max-w-xl bg-white px-28 py-14 shadow-sm rounded-2xl">
						<h4
							className="font-bold text-3xl text-black text-center mb-14">
							Войти
						</h4>

						<label
							className="flex flex-col justify-between font-bold cursor-pointer">
							<span>Email:</span>
							<input
								onChange={this.handleChange}
								value={email}
								type="email"
								name="email"
								placeholder="mail@mail.ru"
								className="mt-1 block w-full border-b-2 border-gray-300 shadow-sm
						focus:outline-none focus:border-yellow-me focus:ring-yellow-me focus:placeholder-yellow-me
						placeholder:text-gray-me"/>
						</label>

						<label className="flex flex-col justify-between font-bold cursor-pointer mt-6">
							<span>Password:</span>
							<input
								onChange={this.handleChange}
								value={password}
								type="password"
								name="password"
								placeholder="*************"
								className="mt-1 block w-full border-b-2 border-gray-300 shadow-sm
						focus:outline-none focus:border-yellow-me focus:ring-yellow-me focus:placeholder-yellow-me
						placeholder:text-gray-me"/>
						</label>

						<p
							className="mt-3 text-right">
							<a
								href="#test"
								className=" text-gray-me cursor-pointer hover:text-yellow-me">
								Забыли пароль?
							</a>
						</p>

						<button
							type="submit"
							className="mt-11 bg-yellow-me w-full py-4 text-2xl rounded-full">
							Войти
						</button>

						<p
							className="mt-8 text-gray-me">
							Новый пользователь?
							<a
								href="#map"
								className="text-yellow-me">
								Регистрация
							</a>
						</p>
					</form>
				</div>
			</section>
		)
	}
}

export default Login