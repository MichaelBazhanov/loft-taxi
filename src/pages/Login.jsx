import React, { useState } from "react";
import logoTaxiVertical from '../assets/images/logo-taxi-vertical.svg';
import { connect } from 'react-redux'
import { authenticate } from '../actions' //просто импортируем action
import PropTypes from "prop-types";
import { useNavigate } from "react-router-dom";

const Login = ({ authenticate, isLoggedIn }) => {
	const [state, setState] = useState({ email: 'test@test.com', password: '123123' }) // hard code

	const navigate = useNavigate()

	const handleSubmit = (event) => {
		event.preventDefault()
		// делаем что то с данными

		const { email, password } = state // получаем из state
		authenticate(email, password) // отдаем данные для авторизации
	}

	const handleChange = (event) => {
		setState({ ...state, [event.target.name]: event.target.value })
	}

	const { email, password } = state

	return (
		<div
			className="container mx-auto flex h-screen bg-map bg-center">
			<div
				className="w-1/3 bg-black-me flex justify-center items-center">
				<img
					src={logoTaxiVertical}
					className="logo-taxi-vertical"
					alt="logo" />
			</div>
			<div
				className="w-2/3 flex justify-center items-center"
			>
				{
					isLoggedIn
						?
						<span
							onClick={() => { navigate("/map", { replace: true }) }}
							className="block text-center cursor-pointer mt-11 bg-yellow-me py-4 px-10 text-2xl rounded-full">
							Перейти на карту
						</span>
						:
						<form
							onSubmit={handleSubmit} className="max-w-xl w-full bg-white px-28 py-14 shadow-lg rounded-2xl ">
							<h4
								className="font-bold text-3xl text-black text-center mb-14">
								Войти
							</h4>

							<label
								className="flex flex-col justify-between font-bold cursor-pointer">
								<span>Email*</span>
								<input
									onChange={handleChange}
									value={email}
									type="email"
									name="email"
									required
									placeholder="mail@mail.ru"
									className="mt-1 block w-full border-b-2 border-gray-300 shadow-sm
						focus:outline-none focus:border-yellow-me focus:ring-yellow-me focus:placeholder-yellow-me
						placeholder:text-gray-me"/>
							</label>

							<label className="flex flex-col justify-between font-bold cursor-pointer mt-6">
								<span>Password*</span>
								<input
									onChange={handleChange}
									value={password}
									type="password"
									name="password"
									required
									placeholder="*************"
									className="mt-1 block w-full border-b-2 border-gray-300 shadow-sm
						focus:outline-none focus:border-yellow-me focus:ring-yellow-me focus:placeholder-yellow-me
						placeholder:text-gray-me"/>
							</label>

							<p
								className="mt-3 text-right">
								<span
									className="inline-block text-gray-me cursor-pointer hover:text-yellow-me">
									Забыли пароль?
								</span>
							</p>

							<button
								type="submit"
								className="block text-center cursor-pointer mt-11 bg-yellow-me w-full py-4 text-2xl rounded-full">
								Войти
							</button>

							<p
								className="mt-8 text-gray-me text-center">
								Новый пользователь?&nbsp;
								<span
									onClick={() => { navigate("/registration") }}
									className="inline-block text-yellow-me cursor-pointer">
									Регистрация
								</span>
							</p>
						</form>
				}

			</div>
		</div>
	)

}

Login.propTypes = {
	authenticate: PropTypes.func.isRequired,
	isLoggedIn: PropTypes.bool.isRequired
}

export default connect(
	(state) => ({ isLoggedIn: state.auth.isLoggedIn }), // ЭТО СЕЛЕКТОР - (state) (auth - это имя reducers) (isLoggedIn - поле)
	{ authenticate } //диспатчим новый экшен
)(Login)

// connect принимает 2 аргумента
// -----------------------------1 селектор
// -----------------------------2 мапинг функции dispatch в props