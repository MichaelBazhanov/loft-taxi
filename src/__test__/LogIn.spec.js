import React from 'react'
import TestRenderer from 'react-test-renderer'
import { Provider } from 'react-redux'

import Login from '../pages/Login'

// Мокаем useNavigate что бы внутри <Login /> отдал просто функцию (ПО СУТИ МЫ МОКАЕМ ХУК)
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => jest.fn(),
}))
// Формируем замоканый стор
const mockStore = {
  getState: () => ({
    // входящие пропсы
    authorizationReducer: { isLoggedIn: true },
    authorizationReducer: { isLoading: false },
    authorizationReducer: { error: false },
  }),
  subscribe: () => {},
  dispatch: () => {
    // диспачим внутри компонента
    authenticate: jest.fn()
  },
}

it('Renders correctly', () => {
  const LoginExample = TestRenderer.create(
    <Provider store={mockStore}>
      <Login />
    </Provider>,
  ).toJSON()

  expect(LoginExample).toMatchSnapshot()
})
