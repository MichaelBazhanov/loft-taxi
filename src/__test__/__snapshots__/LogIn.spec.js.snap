// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div
  className="container mx-auto flex h-screen bg-map bg-center"
>
  <div
    className="w-1/3 bg-black-me flex justify-center items-center"
  >
    <img
      alt="logo"
      className="logo-taxi-vertical"
      src="logo-taxi-vertical.svg"
    />
  </div>
  <div
    className="w-2/3 flex justify-center items-center"
  >
    <form
      autoComplete="off"
      className="max-w-xl w-full bg-white px-28 py-14 shadow-lg rounded-2xl"
      onSubmit={[Function]}
    >
      <h4
        className="font-bold text-3xl text-black text-center mb-14"
      >
        Войти
      </h4>
      <label
        className="flex flex-col justify-between font-bold cursor-pointer"
      >
        <span>
          Email*
        </span>
        <input
          autoComplete="off"
          className="mt-1 block w-full border-b-2 border-gray-300 shadow-sm focus:outline-none focus:border-yellow-me focus:ring-yellow-me focus:placeholder-yellow-me placeholder:text-gray-me"
          name="email"
          onChange={[Function]}
          placeholder="mail@mail.ru"
          required={true}
          type="email"
          value="test@test.com"
        />
      </label>
      <label
        className="flex flex-col justify-between font-bold cursor-pointer mt-6"
      >
        <span>
          Password*
        </span>
        <input
          autoComplete="off"
          className="mt-1 block w-full border-b-2 border-gray-300 shadow-sm focus:outline-none focus:border-yellow-me focus:ring-yellow-me focus:placeholder-yellow-me placeholder:text-gray-me"
          name="password"
          onChange={[Function]}
          placeholder="*************"
          required={true}
          type="password"
          value="123123"
        />
      </label>
      <p
        className="mt-3 text-right"
      >
        <span
          className="inline-block text-gray-me cursor-pointer hover:text-yellow-me"
        >
          Забыли пароль?
        </span>
      </p>
      <button
        className="block text-center cursor-pointer mt-11 bg-yellow-me w-full py-4 text-2xl rounded-full"
        type="submit"
      >
        Войти
      </button>
      <p
        className="mt-8 text-gray-me text-center"
      >
        Новый пользователь? 
        <span
          className="inline-block text-yellow-me cursor-pointer"
          onClick={[Function]}
        >
          Регистрация
        </span>
      </p>
    </form>
  </div>
</div>
`;
